### PR TITLE
RUCIO_Transfers.py: read transfer info from file that need to publish and filter out not-to-publish files from publish container

### PR DESCRIPTION
**Pull request by novicecpp**
Fix https://github.com/dmwm/CRABServer/issues/7741

### CODE REVIEW

**Comment by novicecpp** [comment]
I need to test it before ask Stefano to review.

**Comment by belforte** [comment]
I believe this will work, but hesitate to say "fine" since it looks quite intricated.
Besides what I wrote in comment to the code, I am puzzled that we need another flag.
In Rucio ASO, all files requires transfer, and whether a file should be published or not (i.e. EDM or not) is already part of the file info. Could we simply rely on `outputdatasets.startswith('/FakeDataset` ?

I am trying hard to understand how things work so to get convinced that we can simply set `publish=1` in `transfers.txt. It may requires an "if using Rucio" in the PostJob, but may make things much more readable.

My problem is the usual one, PostJob logic is intricated and I fail to have it clearly in mind while I look at other things.  But this is one reason more to keep `delayed_publicationflag_update` as an internal variable to PostJob.

I know that I approved the original idea of using  `delayed_publicationflag_update` but can not managed to feel good about this.

It is also true that life must go on, so if we do not converge quickly on a better idea, wll have to go for `delayed_publicationflag_update` but should at least add some explanation as comments in the code !

**Review by belforte — CHANGES_REQUESTED** [review]
see comment. Please think about it a bit. If you decide to keep it like this. Fine.

**Comment by novicecpp** [comment]
> I believe this will work, but hesitate to say "fine" since it looks quite intricated. Besides what I wrote in comment to the code, I am puzzled that we need another flag. In Rucio ASO, all files requires transfer, and whether a file should be published or not (i.e. EDM or not) is already part of the file info. Could we simply rely on `outputdatasets.startswith('/FakeDataset` ?

We can set `publish=1` to `transfers.txt`. But, check from `outputdatasets.startswith('/FakeDataset` is better.
Nice. Thanks Stefano.

File: src/python/ASO/Rucio/Transfer.py [hunk 1]
```diff
@@ -139,7 +139,18 @@ def readInfoFromTransferItems(self):
         Convert info from first transferItems to this object attribute.
         Need to execute readTransferItems before this method.
         """
+        # Need to get publish container name from the file that need to publish.
         info = self.transferItems[0]
+        firstJobID = info['job_id']
+        for t in self.transferItems:
+            # break the loop and use first transfers.txt in case there is no
```
[the remark this answers is not in the document]
Comment (novicecpp) [reply src/python/ASO/Rucio/Transfer.py:146]: From what I understand
- We have all output files info to transfer (jobreport + logfile) before postjob execute.
- One Postjob.py process can execute [run()](https://github.com/dmwm/CRABServer/blob/2365939e2777e4afa2077f2c6b485b899c1facde/src/python/TaskWorker/Actions/PostJob.py#L519) at a time.
- [transfers.txt](https://github.com/dmwm/CRABServer/blob/2365939e2777e4afa2077f2c6b485b899c1facde/src/python/TaskWorker/Actions/PostJob.py#L983) get written by [for loop on output files](https://github.com/dmwm/CRABServer/blob/2365939e2777e4afa2077f2c6b485b899c1facde/src/python/TaskWorker/Actions/PostJob.py#L634).

I agree that it is hard to understand and no need to super optimize the code here.

**Comment by belforte** [comment]
> We can set publish=1 to transfers.txt

yes we can, but IIUC only when using Rucio. One day I will have to go through PostJob and document what it does :-(

**Comment by novicecpp** [comment]
We do not use anywhere in FTS_Transfers.py but it is for Publisher to acquire the new files [(ref)](https://github.com/dmwm/CRABServer/blob/2365939e2777e4afa2077f2c6b485b899c1facde/src/python/Databases/FileTransfersDB/Oracle/FileTransfers/FileTransfers.py#L32).

**Review by belforte — APPROVED** [review]

**Comment by novicecpp** [comment]
Change PR title to be more accurate with the changes.

**Pull request merged**